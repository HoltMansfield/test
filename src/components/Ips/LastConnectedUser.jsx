import React from 'react'
import IpActivity from './IpActivity'
import { Heading } from '../styled/Heading'


export const getLastConnectedUser = (ipList) => {
  const sortedByDate = ipList.sort(function(a,b){
    return new Date(b.timestamp) - new Date(a.timestamp);
  })

  return sortedByDate[0]
}

export default function LastConnectedUser({ ipList }) {
  const lastConnectedUser = getLastConnectedUser(ipList)
  return (
      <div>
        <Heading>Most recently connected user</Heading>
        <hr />
        <IpActivity ip={lastConnectedUser} />
      </div>
  )
}
