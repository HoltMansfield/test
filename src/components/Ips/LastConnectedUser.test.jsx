import React from 'react'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LastConnectedUser, { getLastConnectedUser } from './LastConnectedUser'


it('renders correctly', () => {
  const props = {
      ipList: [
        {
          ip: '192.168.0.1',
          timestamp: new Date('January 1, 1990')
        },
        {
          ip: '192.168.0.100',
          timestamp: new Date('January 2, 1991')
        }
      ]
  }
  const wrapper = shallow(<LastConnectedUser {...props} />)

  expect(toJson(wrapper)).toMatchSnapshot()
})

it('getLastConnectedUser returns the most recent visitor', () => {
  const firstUser = {
    ip: '192.168.0.1',
    timestamp: new Date('January 1, 1990')
  }
  const secondUser = {
    ip: '192.168.0.100',
    timestamp: new Date('January 2, 1991')
  }
  const ipList = [firstUser, secondUser]
  const actual = getLastConnectedUser(ipList)

  expect(actual).toEqual(secondUser)
})
