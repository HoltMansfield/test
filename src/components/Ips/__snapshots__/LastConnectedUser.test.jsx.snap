// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <styled.h3>
    Most recently connected user
  </styled.h3>
  <hr />
  <Component
    ip={
      Object {
        "ip": "192.168.0.100",
        "timestamp": 1991-01-02T05:00:00.000Z,
      }
    }
  />
</div>
`;
